app/src/main/java/app/revanced/integrations/youtube/patches/DnsResolverVerificationPatch.java

```diff
@@ -0,0 +1,76 @@
+package app.revanced.integrations.youtube.patches;
+
+import static app.revanced.integrations.shared.StringRef.str;
+
+import android.app.Activity;
+import android.text.Html;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import app.revanced.integrations.shared.Logger;
+import app.revanced.integrations.shared.Utils;
+import app.revanced.integrations.youtube.settings.Settings;
+
+@SuppressWarnings("unused")
+public class DnsResolverVerificationPatch {
+
+    private static final String YOUTUBE_PLAYBACK_HISTORY_API_ADDRESS = "s.youtube.com";
+
+    private static final String SINKHOLE_IPV4 = "0.0.0.0";
```

Comment:
Some DNS blockers may return 127.0.0.1 or any other loopback IP address. Calling the endpoint with HEAD to read a HTTP response code should be more failsafe & doesn't require checking for IPv6 either.